tests/test_browser.py

```diff
@@ -4991,6 +4991,22 @@ def test_wasm2js_fallback(self):
   def test_system(self):
     self.btest(path_from_root('tests', 'system.c'), '0')
 
+  # Tests that it is possible to hook into/override a symbol defined in a system library.
+  @requires_graphics_hardware
+  def test_override_system_js_lib_symbol(self):
+    # This test verifies it is possible to override a symbol from WebGL library.
+
+    # When WebGL is implicitly linked in, the implicit linking should happen before any user --js-libraries, so that they can adjust
+    # the behavior afterwards.
+    self.btest(path_from_root('tests', 'test_override_system_js_lib_symbol.c'),
+               expected='5121',
+               args=['--js-library', path_from_root('tests', 'test_override_system_js_lib_symbol.js')])
+
+    # When WebGL is explicitly linked to in strict mode, the linking order on command line should enable overriding.
+    self.btest(path_from_root('tests', 'test_override_system_js_lib_symbol.c'),
+               expected='5121',
+               args=['-s', 'AUTO_JS_LIBRARIES=0', '-lwebgl.js', '--js-library', path_from_root('tests', 'test_override_system_js_lib_symbol.js')])
```

Comment:
So in this model the library the come last on the command wins?    This seems slightly odd because with native libraries its the first symbol found that wins (reading from left to right).   Not that I think this will be common use case ..

Reply:
That is a good point. The intent of this is to enable overriding built-in JS library functions. The change in this PR only makes it so that the order is preserved, and that JS libraries are processed in the order that is specified.

Currently the mechanism relies on the fact that when you have

```js
// library_a.js
mergeInto(LibraryManager.library, { foo: ... });

// library_b.js
mergeInto(LibraryManager.library, { foo: ... });
```

The `mergeInto` function will cause the overwrite to occur, if library_b.js is linked to after library_a.js.

To switch to the native convention, we'd change `mergeInto` to ignore symbols that are already there (rather than change any code that came in this PR)

However if we did follow the native convention, given that we do have built-in system libraries that are always linked in, we would have a problem that it would not be possible to override any of those built-in functions. That would defeat the purpose of what this PR aims to achieve. So while it would be nice to align with native, I do not propose that we would adhere to that here.

Reply:
sgtm.   I don't think its a big issue.

BTW, in the native linking world the way this works is that the system libraries (libc etc) are always injected *after* any command line arguments from the user.